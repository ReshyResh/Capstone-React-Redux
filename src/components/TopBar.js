import { useSelector } from "react-redux";
import CoinCardsContainer from "./CoinCardsContainer";
import { useState } from "react";
import { useEffect } from "react";

const TopBar = () => {
    useEffect(() => {
        const parent = document.getElementById('top-bar');
        parent.classList.replace('invisible','visible');
      }, []);

    const coins = useSelector((state) => state.coinReducer.coins);
    const gainers = coins.filter((coin) => coin.priceChangePercent > 0);
    const losers = coins.filter((coin) => coin.priceChangePercent < -1);
    gainers.sort((a,b) => (Number(a.priceChangePercent) > Number(b.priceChangePercent) ? 1 : -1));
    losers.sort((a,b) => (Number(a.priceChangePercent) < Number(b.priceChangePercent) ? -1 : 1));

    const [toShow, setToShow] = useState({toShow: coins});

    const handleClick = (selected) => {
        setToShow({toShow: selected})
    }

    return (
        <div id = "top-bar" className="invisible">
            <button onClick = {() => handleClick(coins)}> All </button>
            <button onClick = {() => handleClick(gainers)}> Gainers </button>
            <button onClick = {() => handleClick(losers)}> Losers </button>
            <CoinCardsContainer coins = {toShow.toShow.length === 0 ? coins : toShow.toShow} />
        </div>
        
    )
}

export default TopBar
