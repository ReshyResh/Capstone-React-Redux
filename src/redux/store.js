import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import coinReducer from './reducer/dataReducer';

const reducer = combineReducers({
  coinReducer,
});
const store = createStore(
  reducer,
  applyMiddleware(thunk, logger),
);

export default store;