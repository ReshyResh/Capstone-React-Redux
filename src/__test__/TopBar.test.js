import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import TopBar from '../components/TopBar';
import store from '../redux/store';

it('Top part renders correctly', () => {
  const newDiv = document.createElement('div');
  newDiv.setAttribute('id', 'top-bar');
  document.body.appendChild(newDiv);
  const tree = renderer
    .create(<Provider store={store}><TopBar /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
