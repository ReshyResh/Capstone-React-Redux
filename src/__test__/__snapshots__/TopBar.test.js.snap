// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top part renders correctly 1`] = `
<div
  id="top-bar"
>
  <div
    className="intro white"
  >
    <button
      className="btn btn-outline-light"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "position": "absolute",
          "right": "4%",
          "top": "3%",
          "width": "40px",
          "zIndex": "99",
        }
      }
      type="button"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <line
          x1="12"
          x2="12"
          y1="16"
          y2="12"
        />
        <line
          x1="12"
          x2="12.01"
          y1="8"
          y2="8"
        />
      </svg>
    </button>
    <h1>
       Hello, welcome to my cryptocurrency tracker! 
    </h1>
    <h3>
      Sort coins by:
    </h3>
    <div
      className="sort-container"
    >
      <button
        className="sort-button"
        id="all-button"
        onClick={[Function]}
        type="button"
      >
         
        Show all
         
      </button>
      <button
        className="sort-button"
        id="gainers-button"
        onClick={[Function]}
        type="button"
      >
         
        Top Gainers
         
      </button>
      <button
        className="sort-button"
        id="losers-button"
        onClick={[Function]}
        type="button"
      >
         
        Top Losers
         
      </button>
      <svg
        fill="none"
        height="100"
        id="frame1"
        viewBox="0 0 200 100"
        width="200"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clipPath="url(#clip0)"
          id="Frame 1"
        >
          <path
            d="M96.3707 48.0538C97.3623 47.4125 98.6377 47.4125 99.6293 48.0538L157.883 85.731C160.393 87.3547 159.243 91.25 156.254 91.25L39.7464 91.25C36.7565 91.25 35.6066 87.3547 38.1171 85.731L96.3707 48.0538Z"
            id="tr-3"
          />
          <path
            d="M96.348 27.0898C97.3502 26.4287 98.6499 26.4287 99.652 27.0898L158.072 65.6281C160.556 67.2668 159.396 71.1324 156.42 71.1324H39.5798C36.604 71.1324 35.4438 67.2668 37.9278 65.6281L96.348 27.0898Z"
            id="tr-2"
          />
          <path
            d="M97.8847 1.09039C98.887 0.428982 100.187 0.42898 101.189 1.09039L158.346 38.8078C160.829 40.4467 159.669 44.3118 156.693 44.3118H42.3809C39.4054 44.3118 38.245 40.4467 40.7285 38.8078L97.8847 1.09039Z"
            id="tr-1"
          />
        </g>
        <defs>
          <clipPath
            id="clip0"
          >
            <rect
              fill="white"
              height="100"
              width="200"
            />
          </clipPath>
        </defs>
      </svg>
      <svg
        fill="none"
        height="100"
        id="frame2"
        viewBox="0 0 200 100"
        width="200"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clipPath="url(#clip0)"
          id="Frame 1"
        >
          <path
            d="M100.5 3L112.961 38.5856H153.284L120.662 60.5788L133.122 96.1644L100.5 74.1712L67.8779 96.1644L80.3384 60.5788L47.7164 38.5856H88.0395L100.5 3Z"
            id="star"
          />
        </g>
        <defs>
          <clipPath
            id="clip0"
          >
            <rect
              fill="white"
              height="100"
              width="200"
            />
          </clipPath>
        </defs>
      </svg>
    </div>
  </div>
  <div
    className="card-container"
  />
</div>
`;
